### PR TITLE
dtls.c: introduce flags parameter for dtls_destroy_peer()

### PR DESCRIPTION
**Pull request by obgm**
The parameter unlink for dtls_destroy_peer() has been in use for several years and therefore can be safely removed as suggested in PR #154.

### CODE REVIEW

**Comment by boaks** [comment]
I would prefer to "reuse" it for the close.

**Comment by obgm** [comment]
The fixup now introduces a flag to control sending of CloseNotify when applicable (i.e., not if the peer state is CLOSED or CLOSING).

File: dtls.c [hunk 1]
```diff
@@ -62,6 +62,13 @@
 
 #define DTLS10_VERSION 0xfeff
 
+/* Flags for dtls_destroy_peer()
+ *
+ *  DTLS_DESTROY_CLOSE indicates that the connection should be closed
+ *                     when applicable
+ */
+#define DTLS_DESTROY_CLOSE 0x02
```
Comment (boaks) [inline dtls.c:70]: That's a nice idea.

In a future PR, I would then go to add

```
DTLS_DESTROY_WITHOUT_CLOSE 0x01
```

**Comment by obgm** [comment]
Great, the squashed version now goes into main.

**Pull request merged**